coding/url.hpp

```diff
@@ -37,7 +39,7 @@ class Url
   std::string const * GetParamValue(std::string const & name) const
   {
     for (auto const & p : m_params)
-      if (p.first == name)
+      if (boost::iequals(p.first, name))
```

Comment:
Hm. You need to compare with lowercase or what?
Anyway, I'd not use boost here. Better something like this:
```
name.size() == p.first.size() &&
std::equal(name.begin(), name.end(), p.first.begin(), [](char c1, char c2)
{
  return std::tolower(c1) == std::tolower(c2);
});
```

Since url is always ASCII (or not?), I'd make some custom strings::ToLower(char) function like in strings::AsciiToLower

Reply:
This is not because of boost, but in C++ you can't rely on strings processing, because of complicated locale routine. Better to do _explicitly_ what you want to do :)

Reply:
I want to compare two strings case insensitively. This is a very simply operation that exists in almost any standard library, but not in C++.

> Better something like this:

```
name.size() == p.first.size() &&
std::equal(name.begin(), name.end(), p.first.begin(), [](char c1, char c2)
{
  return std::tolower(c1) == std::tolower(c2);
});
```

Yeah, this is definitely better and easy to read.

Reply:
What about cyrillic (or other) URLs that are not percent encoded? Did you test them?
Why case-insensitive comparison is needed at all? Is it critical for the code to work? If yes, I suggest to use [UnicodeString::caseCompare](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1UnicodeString.html#a65aca3fd2b1d9156f7966c50dacdcffe)

Reply:
URL param's names are always ASCII. You can add strings::EqualAsciiLowerCase function and put my code there.

Reply:
How about this?

https://omaps.app/map?v=1&ll=22.17319,-159.65687&n=Калалау%20Camping&ll=22.17168,-159.66096&n=Пляж%20мечты&ll=22.17182,-159.65776

Reply:
Param names, not values here. Like n, z, ll, map.

Reply:
Query string parameters in URI are not case-insensitive. I've removed this change from the patch. Nothing to discuss here.

Reply:
Hm, boost::iequals is still present

Reply:
oops, bad rebase! I will remove.